src/rasterstats/main.py

```diff
@@ -197,8 +198,12 @@ def gen_zonal_stats(
             else:
                 if run_count:
                     keys, counts = np.unique(masked.compressed(), return_counts=True)
-                    pixel_count = dict(zip([np.asscalar(k) for k in keys],
-                                           [np.asscalar(c) for c in counts]))
+                    if LooseVersion(np.__version__) < LooseVersion('1.16.0'):
+                        pixel_count = dict(zip([np.asscalar(k) for k in keys],
+                                               [np.asscalar(c) for c in counts]))
+                    else:
+                        pixel_count = dict(zip([k.item() for k in keys],
+                                               [c.item() for c in counts]))
```

Comment:
I'd like to avoid explicit version checks if possible. In other words, the code could check for the presence of the `item()` method and use it if available, only falling back on `np.asscalar` if absolutely needed.

The item method is slightly faster so this will provide a small speed improvement as well

```
In [1]: a = np.array(range(1000000))

In [2]: %timeit [x.item() for x in a]
534 ms ± 2.33 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

In [3]: %timeit [np.asscalar(x) for x in a]
693 ms ± 3.33 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```

Reply:
Glad to hear that it's faster. Do you think wrapping the functions into a `try` and `except` test would be adequate? This would also remove the need for version checking.

Reply:
Looks good to me!